File: prolog/eliuds-eggs/eliuds_eggs.pl
:- use_module(library(clpfd)).

egg_count(0, 0) :- !.
egg_count(EncodedCount, Count) :-
    NewEncodedCount #= EncodedCount >> 1,
    egg_count(NewEncodedCount, NewCount),
    Count #= NewCount + (EncodedCount /\ 1).

File: r/eliuds-eggs/eliuds-eggs.R
egg_count <- function(display_value) {
  count <- 0
  
  while (display_value > 0) {
    count <- count + bitwAnd(display_value, 1)
    display_value <- bitwShiftR(display_value, 1)
  }
  
  count
}
